core/connection/manager.go

```diff
@@ -277,15 +282,15 @@ func warnOnClean() {
 	log.Warn(managerLogPrefix, "Trying to close when there is nothing to close. Possible bug or race condition")
 }
 
-func connectionWaiter(connection Connection, dialog communication.Dialog) {
+func (manager *connectionManager) connectionWaiter(connection Connection) {
 	err := connection.Wait()
 	if err != nil {
 		log.Warn(managerLogPrefix, "Connection exited with error: ", err)
 	} else {
 		log.Info(managerLogPrefix, "Connection exited")
 	}
 
-	dialog.Close()
+	_ = manager.Disconnect()
```

Comment:
same here.. afraid of underscores ;)

Reply:
The only error we're possibly returning from disconnect is the `ErrNoConnection`, which is going to flood the logs needlessly, since these disconnects will get triggered by multiple goroutines(possibly).

Reply:
I've added a method to log the error and ignore the ErrNoConnection